sessions/forecasting-models.qmd

```diff
@@ -31,6 +31,631 @@ The best way to interact with the material is via the [Visual Editor](https://do
+The worst case is that we have mistaken beliefs about the underlying mechanisms and use these to make predictions which are systematically wrong and misleading.
+In these cases, forecasters of often use _statistical models_ to make predictions which have little or no mechanistic basis.
+In our work, we have found that a combination of mechanistic and statistical models can be very powerful but that identifying the best model for a given forecasting task can be challenging.
+
+## Adding more mechanistic structure to the renewal model
+
+One way to improve the renewal model is to add more mechanistic structure. In the [forecasting concepts session](forecasting-concepts), we saw that the renewal model was unbiased when the reproduction number was constant but that it overestimated the number of cases when the reproduction number was reducing due to susceptible depletion.
+
+This suggests that we should add a term to the renewal model which captures the depletion of susceptibles. One way to do this is to add a term which is proportional to the number of susceptibles in the population. This is the idea behind the _SIR model_ which is a simple compartmental model of infectious disease transmission. If we assume that susceptible depletion is the only mechanism which is causing the reproduction number to change, we can write the reproduction model as:
+
+$$
+R_t = \frac{S_{t-1}}{N} R_0
+$$
+
+This approximates susceptible depletion as a linear function of the number of susceptibles in the population. This is a simplification but it is a good starting point.
+
+::: {.callout-note, collapse = TRUE}
+## What behaviour would we expect from this model?
+
+```{r}
+set.seed(123)
+n <- 100
+N <- 1000
+R0 <- 1.5
+S <- rep(NA, n)
+S[1] <- N
+R <- rep(NA, n)
+R[1] <- R0
+I <- rep(NA, n)
+I[1] <- 1
+for (i in 2:n) {
+  R[i] <- (S[i-1]) / N * R0
+  I[i] <- I[i-1] * R[i]
+  S[i] <- S[i-1] - I[i]
+}
+
+data <- tibble(t = 1:n, R = R)
+
+ggplot(data, aes(x = t, y = R)) +
+  geom_line() +
+  labs(title = "Simulated data from an SIR model",
+       x = "Time",
+       y = "R")
+```
+
+The key assumptions we are making here are: 
+
+- The population is constant and we rouchgly know the size of the population.
+- The reproduction number is constant (aside from the effect of susceptible depletion)
+- The number of cases is proportional to the number of susceptibles in the population.
+
+We've coded this up as a stan model in `stan/mechanistic-r.stan`. See `stan/functions/pop_bounded_renewal.stan` for the function which calculates the reproduction number. Lets load the model:
+
+```{r}
+mech_mod <- cmdstan_model(here("stan/mechanistic-r.stan"))
+mech_mod$print(line_numbers = TRUE)
+```
+
+## Adding more statistical structure to the renewal model
+
+Rather than adding more mechanistic structure to the renewal model, we can add more statistical structure.
```

Comment:
```suggestion
Adding more mechanistic structure is not always possible and, if we don't specify mechanisms correctly, might make forecasts worse.
Rather than adding more mechanistic structure to the renewal model, we could add more statistical structure.
```